CMakeLists.txt

```diff
@@ -173,6 +173,12 @@ if(MSVC)
     string(REPLACE "/GR " "/GR- " CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS})
     string(REPLACE "/GR " "/GR- " CMAKE_CXX_FLAGS_DEBUG ${CMAKE_CXX_FLAGS_DEBUG})
     string(REPLACE "/GR " "/GR- " CMAKE_CXX_FLAGS_RELEASE ${CMAKE_CXX_FLAGS_RELEASE})
+
+    # Compiling with guard:ehcont allow library consumers to enable guard:ehcont if they want. (Requires VS 2019 x64)
+    if((MSVC_VERSION GREATER 1920) AND (CMAKE_SIZEOF_VOID_P EQUAL 8))
```

Comment:
Make this ``1928`` since this requires VS 2019 16.7 or later.

Reply:
Also, what happens here when building for ARM64? Also the target name here is wrong, so I'm guessing this was a cut & paste.

This should be moved down to the ``if(MSVC)`` section with other such conditionals.

```
if(MSVC)
    foreach(t IN LISTS TOOL_EXES ITEMS ${PROJECT_NAME})
      target_compile_options(${t} PRIVATE /fp:fast "$<$<NOT:$<CONFIG:DEBUG>>:/guard:cf>")
      target_link_options(${t} PRIVATE /DYNAMICBASE /NXCOMPAT)
    endforeach()

    if((${CMAKE_SIZEOF_VOID_P} EQUAL 4) AND (NOT ${DIRECTX_ARCH} MATCHES "^arm"))
        foreach(t IN LISTS TOOL_EXES ITEMS ${PROJECT_NAME})
          target_link_options(${t} PRIVATE /SAFESEH)
        endforeach()
    endif()

    if((MSVC_VERSION GREATER_EQUAL 1928) AND (${CMAKE_SIZEOF_VOID_P} EQUAL 8) AND (NOT ${DIRECTX_ARCH} MATCHES "^arm"))
        foreach(t IN LISTS TOOL_EXES ITEMS ${PROJECT_NAME})
          target_link_options(${t} PRIVATE "$<$<CONFIG:RELEASE>:/guard:ehcont>")
        endforeach()
    endif()
endif()
```

Reply:
Documentation says that /guard:ehcont is "supported for 64-bit processes on a 64-bit OS" so I'm assuming ARM64 is also supported.